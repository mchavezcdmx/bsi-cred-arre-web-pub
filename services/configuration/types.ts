const TYPES = {
  ContentfulRepository: Symbol.for('ContentfulRepository'),
  ExampleRepository: Symbol.for('ExampleRepository')
};

export { TYPES };
