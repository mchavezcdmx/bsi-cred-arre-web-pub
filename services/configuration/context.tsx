import React, {
  createContext,
  useContext
} from 'react';
import { Container } from 'inversify';
import { repositoryContainer } from './inversify.config';

const RepositoryIocContext = createContext(null);

export const RepositoryIocProvider: React.FC<{ container?: Container }> = ({
  container,
  children
}) => {
  const value = { container: container || repositoryContainer };

  return <RepositoryIocContext.Provider value={value}>{children}</RepositoryIocContext.Provider>;
};

export const useRepositoryIoc = () => useContext(RepositoryIocContext);
