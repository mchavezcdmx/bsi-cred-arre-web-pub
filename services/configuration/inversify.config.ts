// eslint-disable-next-line sort-imports
import 'reflect-metadata';

import { Container } from 'inversify';
import { ContentfulRepository } from '../repositories/contenful.repository';
import { ContentfulRepositoryImpl } from '../repositories/contenful.repository.impl';
import { ExampleRepository } from '../repositories/example.repository';
import { ExampleRepositoryMock } from '../repositories/example.repository.mock';
import { TYPES } from './types';

const repositoryContainer = new Container();

repositoryContainer.bind<ExampleRepository>(TYPES.ExampleRepository).to(ExampleRepositoryMock);
repositoryContainer.bind<ContentfulRepository>(TYPES.ContentfulRepository).to(ContentfulRepositoryImpl);

export { repositoryContainer };
