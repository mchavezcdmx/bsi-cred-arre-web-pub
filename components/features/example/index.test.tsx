import {
  AnyAction, Dispatch, Middleware
} from 'redux';
import configureStore, { MockStoreCreator } from 'redux-mock-store';
import {
  decrement, increment
} from '../../../redux/reducers/counter.slice';
import {
  fireEvent, render, screen
} from '@testing-library/react';
import {
  QueryClient, QueryClientProvider
} from 'react-query';
import Example from '.';
import { Provider } from 'react-redux';
import { RepositoryIocProvider } from '../../../services/configuration/context';

const middlewares: Middleware<{}, any, Dispatch<AnyAction>>[] = [];
const mockStore: MockStoreCreator<unknown, {}> = configureStore(middlewares);

describe('Test in example component', () => {
  const initialState = { counter: { value: 0 } };

  const queryClient = new QueryClient();

  const store = mockStore(initialState);

  store.dispatch = jest.fn();

  test('Should render the component', () => {
    const { asFragment } = render(
      <RepositoryIocProvider>
        <Provider store={store}>
          <QueryClientProvider client={queryClient}>
            <Example />
          </QueryClientProvider>
        </Provider>
      </RepositoryIocProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('should dispatch increment on click on increment button', () => {
    render(
      <RepositoryIocProvider>
        <Provider store={store}>
          <QueryClientProvider client={queryClient}>
            <Example />
          </QueryClientProvider>
        </Provider>
      </RepositoryIocProvider>
    );

    const incrementButton = screen.getByText(/Increment/);

    fireEvent.click(incrementButton);

    expect(store.dispatch).toHaveBeenCalledWith(increment());
  });
  test('should dispatch decrement on click on decrement button', () => {
    render(
      <RepositoryIocProvider>
        <Provider store={store}>
          <QueryClientProvider client={queryClient}>
            <Example />
          </QueryClientProvider>
        </Provider>
      </RepositoryIocProvider>
    );

    const decrementButton = screen.getByText(/Decrement/);

    fireEvent.click(decrementButton);

    expect(store.dispatch).toHaveBeenCalledWith(decrement());
  });
});
