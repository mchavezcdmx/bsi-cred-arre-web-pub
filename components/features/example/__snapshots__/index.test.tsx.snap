// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in example component Should render the component 1`] = `
<DocumentFragment>
  <div
    class="example"
  >
    <h1 />
    <div>
      <button
        aria-label="Increment value"
      >
        Increment
      </button>
      <p>
        0
      </p>
      <button
        aria-label="Decrement value"
      >
        Decrement
      </button>
    </div>
  </div>
</DocumentFragment>
`;
